crates/krator/src/admission.rs

```diff
@@ -0,0 +1,115 @@
+//! Basic implementation of Kubernetes Admission API
+use crate::Operator;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::Status;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+/// Result of admission hook.
+#[allow(clippy::large_enum_variant)]
+pub enum AdmissionResult<T> {
+    /// Permit the request. Pass the object (with possible mutations) back.
+    /// JSON Patch of any changes will automatically be created.
+    Allow(T),
+    /// Deny the request. Pass a Status object to provide information about the error.
+    Deny(Status),
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+/// AdmissionRequest describes the admission.Attributes for the admission request.
+struct AdmissionRequest<T> {
```

Comment:
Should all of these types be available in the `k8s_openapi` crate or are they something technically outside of the normal API spec?

Reply:
I'm not totally sure, they are defined [here](https://github.com/kubernetes/api/blob/master/admission/v1/generated.proto) which is in the main API source tree, but I searched for a while and could not find anything in `k8s_openapi` that implemented this. As far as I can tell it may be because there isn't an OpenAPI spec published for this by the Kubernetes API server.